crates/ide_assists/src/handlers/generate_function.rs

```diff
@@ -79,6 +100,67 @@ pub(crate) fn generate_function(acc: &mut Assists, ctx: &AssistContext) -> Optio
     )
 }
 
+pub(crate) fn generate_method(acc: &mut Assists, ctx: &AssistContext) -> Option<()> {
+    let fn_name: ast::NameRef = ctx.find_node_at_offset()?;
+    let call: ast::MethodCallExpr = ctx.find_node_at_offset()?;
+    let ty = ctx.sema.type_of_expr(&call.receiver()?)?.original().strip_references().as_adt()?;
+
+    let current_module =
+        ctx.sema.scope(ctx.find_node_at_offset::<ast::MethodCallExpr>()?.syntax()).module()?;
+    let target_module = ty.module(ctx.sema.db);
+
+    if current_module.krate() != target_module.krate() {
+        return None;
+    }
+
+    let (impl_, file) = match ty {
+        hir::Adt::Struct(strukt) => get_impl(strukt.source(ctx.sema.db)?.syntax(), &fn_name, ctx),
+        hir::Adt::Enum(en) => get_impl(en.source(ctx.sema.db)?.syntax(), &fn_name, ctx),
+        hir::Adt::Union(union) => get_impl(union.source(ctx.sema.db)?.syntax(), &fn_name, ctx),
+    }?;
+
+    let function_builder = FunctionBuilder::from_method_call(
+        ctx,
+        &call,
+        &fn_name,
+        &impl_,
+        file,
+        target_module,
+        current_module,
+    )?;
+    let target = call.syntax().text_range();
+
+    acc.add(
+        AssistId("generate_method", AssistKind::Generate),
+        format!("Generate `{}` method", function_builder.fn_name),
+        target,
```

Comment:
I think you want me to change the `generate_method` to `generate_function`, right?